diff_cover/violationsreporters/violations_reporter.py

```diff
@@ -384,33 +384,48 @@ def parse(lcov_file):
                     line_no, _ = function_lines[source_file][func_name]
                     function_lines[source_file][func_name] = (line_no, hit_count)
             elif directive in [
-                "TN",
-                "FNF",
-                "FNH",
-                "LH",
-                "LF",
-                "BRF",
-                "BRH",
-                "VER",
+                "TN",  # Test name
+                "FNF",  # Functions found
+                "FNH",  # Functions hit
+                "LH",  # Lines hit
+                "LF",  # Lines found
+                "BRF",  # Branches found
+                "BRH",  # Branches hit
+                "VER",  # Version
+                "FNL",  # Function line coverage (alternative format)
+                "FNA",  # Function name (alternative format)
             ]:
-                # these are valid lines, but not we don't need them
+                # Valid directives that we don't need to process
                 continue
             elif directive == "end_of_record":
-                # Apply branch coverage filtering
-                for line_no, info in branch_coverage[source_file].items():
-                    if line_no not in lcov_report[source_file]:
-                        raise ValueError(
-                            f"Line {line_no} not found in lcov report for {source_file}"
-                        )
-                    # Only keep lines that all branches are hit
-                    # TODO: add an option to keep lines with partial branch coverage
-                    if info["total"] > 0 and info["hit"] < info["total"]:
-                        lcov_report[source_file][line_no] = 0
+                # Process collected coverage data for current source file
 
-                # Apply function hit filtering
+                # 1. Apply branch coverage logic
+                for line_no, info in branch_coverage[source_file].items():
+                    has_da_directive = line_no in lcov_report[source_file]
+
+                    if not has_da_directive:
+                        # No line execution data, use branch coverage
+                        if info["total"] > 0 and info["hit"] < info["total"]:
+                            lcov_report[source_file][
+                                line_no
+                            ] = 0  # Partial branch coverage
+                        else:
+                            lcov_report[source_file][line_no] = info["executions"]
+                    elif not lcov_report[source_file][line_no]:
+                        # Line shows 0 executions, but check if branches were hit
+                        if info["executions"] > 0:
+                            lcov_report[source_file][line_no] = info["executions"]
+                    # Note: Don't override existing positive execution counts
```

Comment:
I dont see how these blocks are related to each other
```suggestion
                    if not lcov_report[source_file][line_no]:
                        # Line shows 0 executions, but check if branches were hit
                        if info["executions"] > 0:
                            lcov_report[source_file][line_no] = info["executions"]
                    # Note: Don't override existing positive execution counts
```